src/main/scala/com/microsoft/ml/spark/automl/FindBestModel.scala

```diff
@@ -176,7 +180,10 @@ class BestModel(val uid: String) extends Model[BestModel]
 
   override def copy(extra: ParamMap): BestModel = defaultCopy(extra)
 
-  override def transform(dataset: Dataset[_]): DataFrame = getBestModel.transform(dataset)
+  override def transform(dataset: Dataset[_]): DataFrame = {
+    logTransform(uid)
```

Comment:
could this be achieved using the [Mixin Composition Stacks](http://jonasboner.com/aop-style-mixin-composition-stacks-in-scala/)

Reply:
Do you mean that we could override the transform function by extending the trait/class? I log all transform functions for different classes so it looks not applicable to extend them one by one. Pls correct me if I didn't get you!

Reply:
I think it's worth a try. if you define override def transform(dataset: Dataset[_]): DataFrame = { ... }  in the Loggable trait, which method is called first? The Loggable or the FindBestModel? if it's the Loggable trait, I think you won, as you don't need the individual logTransform(uid) calls...